eth/tracers/api.go

```diff
@@ -266,6 +266,7 @@ func (api *API) traceChain(ctx context.Context, start, end *types.Block, config
 				// Trace all the transactions contained within
 				for i, tx := range task.block.Transactions() {
 					msg, _ := tx.AsMessage(signer)
+					task.statedb.Prepare(tx.Hash(), task.block.Hash(), i)
 					res, err := api.traceTx(ctx, msg, blockCtx, task.statedb, config)
```

Comment:
I'd be a lot happier if we did thins in traceTx, seems wried to manually do this at all callsites

Reply:
Ok, but we don't actually have the index nor the hash at that point. That may be fine, but might also at some later point pop up as as bug